webserver/script/CostBenefitAnalysis/python_code/loadprofile_class.py

```diff
@@ -0,0 +1,258 @@
+        """
+
+        for year in model_years:
+
+            try:
+                weekday_count = {i: 0 for i in range(24)}
+                weekday_sum = {i: 0 for i in range(24)}
+
+                weekend_count = {i: 0 for i in range(24)}
+                weekend_sum = {i: 0 for i in range(24)}
+
+                for timestep in list(timesteps.keys()):
+                    is_weekday = timesteps[timestep][year]['is_weekday']
+                    hour_of_day = timesteps[timestep][year]['hourofday']
+                    load = self.annual_load[year][timestep]
+
+                    if is_weekday:
+                        weekday_count[hour_of_day] += 1
+                        weekday_sum[hour_of_day] += load
+                    else:
+                        weekend_count[hour_of_day] += 1
+                        weekend_sum[hour_of_day] += load
+
+                self.avg_weekday[year] = {i: old_div(weekday_sum[i], weekday_count[i]) for i in range(24)}
+                self.avg_weekend[year] = {i: old_div(weekend_sum[i], weekend_count[i]) for i in range(24)}
+
+            except KeyError:
+                continue
+
+    def get_peak_day(self, timesteps, model_years):
+
+        for year in model_years:
+
+            try:
+                peak_timestep = helpers.get_max_index(self.annual_load[year])
+
+                month = timesteps[peak_timestep][year]['month']
+                dayofmonth = timesteps[peak_timestep][year]['dayofmonth']
+
+                peak_timesteps = sorted([i for i in list(timesteps.keys()) if timesteps[i][year]['month'] == month
+                    and timesteps[i][year]['dayofmonth'] == dayofmonth])
+
+                self.peak_shape[year] = {i: self.annual_load[year][peak_timesteps[i]] for i in range(24)}
+                self.peak_isweekday[year] = timesteps[peak_timesteps[0]][year]['is_weekday']
+
+            except KeyError:
+                continue
+
+
+    def expand_loadprofiles(self, model_years, vehicles):
+        """
+        Given the annual_load attribute, model_years and a vehicle population over time, expand_loadprofiles() will
+        expand the annual_load attribute into the missing model years by scaling based on either the first or last
+        year in annual_load.
+        """
+
+        first_loadprofile_year = min(self.annual_load.keys())
+        last_loadprofile_year = max(self.annual_load.keys())
+
+        for year in model_years:
+
+            if year not in list(self.annual_load.keys()):
+                pop_in_year = vehicles.population[year]
+
+                if year < first_loadprofile_year:
+                    template_year = first_loadprofile_year
+                else:
+                    template_year = last_loadprofile_year
+
+                pop_in_loadprofile = vehicles.population[template_year]
+                try:
+                    scalar = old_div(pop_in_year,pop_in_loadprofile)
+                except:

```

Comment:
Do we know what exactly we should be catching here?

Reply:
I do not. The cba_tool is supposed to be seen as a "black box". I know Robbie is also in the process of writing a new version of the CBA tool as well.